ticdc panic when a changefeed with redo log enabled is paused

### What did you do?

1. setup a ticdc replication, with redo log enabling 
2. pause the changefeed

### What did you expect to see?

_No response_

### What did you see instead?

```
panic: send on closed channel

goroutine 150161 [running]:
github.com/pingcap/ticdc/cdc/redo.(*ManagerImpl).bgUpdateResolvedTs(0xc006bc3c20, 0x3b3c5d0, 0xc01011a0f0, 0xc00c1902a0)
        github.com/pingcap/ticdc/cdc/redo/manager.go:346 +0x175
created by github.com/pingcap/ticdc/cdc/redo.NewManager
        github.com/pingcap/ticdc/cdc/redo/manager.go:194 +0x589
```

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
v5.2.1
```

TiCDC version (execute `cdc version`):

master@https://github.com/pingcap/ticdc/commit/628b5d610e9c7411e6394ff1318308f3482c6f42

--- a/cdc/processor/processor.go
+++ b/cdc/processor/processor.go
@@ -225,6 +225,9 @@ func (p *processor) lazyInitImpl(ctx cdcContext.Context) error {
 	ctx, cancel := cdcContext.WithCancel(ctx)
 	p.cancel = cancel
 
+	// We don't close this error channel, since it is only safe to close channel
+	// in sender, and this channel will be used in many modules including sink,
+	// redo log manager, etc. Let runtime GC to recycle it.
 	errCh := make(chan error, 16)
 	p.wg.Add(1)
 	go func() {
@@ -236,7 +239,6 @@ func (p *processor) lazyInitImpl(ctx cdcContext.Context) error {
 		for {
 			select {
 			case <-ctx.Done():
-				close(errCh)
 				return
 			case err := <-errCh:
 				if err == nil {

